### PR TITLE
Add function to correctly format a DS string

### PR DESCRIPTION
**Pull request by CPBridge**
According to the standard, attributes with the decimal string (DS) value representation must contain a maximum of 16 characters (see [here](http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html)). Unfortunately pydicom does not enforce this, and it looks like there is no movement on @hackermd's [open issue](https://github.com/pydicom/pydicom/issues/1264) to rectify this.

A quick fix was previously implemented in the `PlanePositionSequence`, but this was sub-optimal as it truncated (rather than correctly rounding) and did not handle certain cases correctly (e.g. values that python would write in scientific notation when calling the str() method on a float).

Furthermore, there are several other instances around the codebase where this fix was not implemented, meaning that invalid DS could be created. For example in NumContentItem, which is used for recording nearly all numeric values in an SR document. I found that this was causing many SRs I created to fail the checks in the dciodvfy tool.

This PR implements a new function that I believe correctly handles all edge cases, and rounds properly. It also checks for nans/infs, which would previously be silently stored as strings despite being invalid values for DS. Also added tests and docs.

I have tried to find all the places in the codebase where DS elements are created and switch to using the new function, but I'm not 100% sure I found that I found them all.

### CODE REVIEW

File: src/highdicom/valuerep.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+from math import log10, floor
+
+from numpy import isfinite
+
+
+def get_ds_string(f: float) -> str:
```
Comment (pieper) [inline src/highdicom/valuerep.py:6]: Wouldn't it make sense for this to be in pydicom instead?

Comment (hackermd) [reply src/highdicom/valuerep.py:6]: @pieper I agree, this would ideally go into pydicom. There is an option issue (see https://github.com/pydicom/pydicom/issues/1264). @CPBridge have you considered creating a PR to update `pydicom.valuerep.DS`?

**Review by hackermd — CHANGES_REQUESTED** [review]
Thanks @CPBridge. This looks great. However, I agree with @pieper that this would ideally go into pydicom. Is this an option?

File: src/highdicom/content.py [hunk 2]
```diff
@@ -261,7 +265,7 @@ def __eq__(self, other: Any) -> bool:
         Returns
         -------
         bool
-            Whether the two image planes have the same orientation
+            Whether the two image planes ha[
```
[the remark this answers is not in the document]
Comment (CPBridge) [reply src/highdicom/content.py]: Ooops

**Comment by CPBridge** [comment]
I would certainly agree with @hackermd  and @pieper that this should be fixed at the pydicom level and this was my first thought also, but given this response in the [existing issue](https://github.com/pydicom/pydicom/issues/1264) in pydicom:

> "...pydicom should leave it to the calling code to figure out how to truncate the value."

I got the impression that this would not be welcomed.

However I suppose I night be reading too much into this and any code that does not force pydicom users to truncate their numbers in a certain way might still be welcomed. I could try adding it as a utility function in pydicom that users of pydicom can choose to use explicitly.

Let's park this PR here for now and I will submit a version to pydicom and see where that leads.

**Comment by hackermd** [comment]
> However I suppose I night be reading too much into this and any code that does not force pydicom users to truncate their numbers in a certain way might still be welcomed. I could try adding it as a utility function in pydicom that users of pydicom can choose to use explicitly.

Handling of DS is tricky and can cause really nasty bugs. I understand that we need to workaround compliance issues, but it shouldn't go as far as making it hard for people to actually create valid DICOM in the first place.

An approach as proposed in https://github.com/MGHComputationalPathology/highdicom/issues/56#issuecomment-810679595 may help with that.

**Comment by CPBridge** [comment]
My approach here is to:
- Get the functionality for correctly truncating a float into a 16 char string into pydicom, but have it entirely optional for users of pydicom
- Have users of highdicom continue to pass floats into the highdicom API, and then call the truncation routine from pydicom whenever the attribute is set within highdicom code (the same places I updated in the original PR)

Sound like a good approach?

**Comment by hackermd** [comment]
> Sound like a good approach?

Sounds good to me. @darcymason what do you think? Would that be a good compromise?

**Comment by darcymason** [comment]
> Sounds good to me. @darcymason what do you think? Would that be a good compromise?

Given that it is proposed as an explicit call, I can't see any problem with that.

**Comment by CPBridge** [comment]
See https://github.com/pydicom/pydicom/pull/1334

**Comment by CPBridge** [comment]
This functionality has just been merged into pydicom master branch https://github.com/pydicom/pydicom/pull/1334 and should be in the next release of pydicom (looks like this will be 2.2.0). I will adapt this PR to use that function instead

File: src/highdicom/valuerep.py (hunk 1 above)
Comment (hackermd) [reply src/highdicom/valuerep.py:6]: @CPBridge given that this has been merged into pydicom, can we remove this module?

File: tests/test_valuerep.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import pytest
```
Comment (hackermd) [inline tests/test_valuerep.py:1]: @CPBridge I assume we no longer need these tests in highdicom, do we?

Comment (CPBridge) [reply tests/test_valuerep.py:1]: No this PR was originally written with the assumption that the functionality would be in highdicom and I haven't adapted it to at all yet for the fact that the functionality has moved to pydicom. A lot will need to change. I might just create a new PR from scratch, not sure which will end up being easier

File: src/highdicom/valuerep.py (hunk 1 above)
Comment (CPBridge) [reply src/highdicom/valuerep.py:6]: yes, there's not sense reviewing this PR in it's current form

**Comment by CPBridge** [comment]
Closing in favour of #65
